src/write_to_file.cc

```diff
@@ -47,20 +51,22 @@ int writeValueToFile(
 }
 
 
-void writeVectorToFile(std::vector<double> v, std::string pFilename){
+void writeVectorToFile(std::vector<double> v, fs::path filename){
```

Comment:
I thought about that. But there are some objections
 - if we change the filename internally, writing "const" is confusing for the user (user writes filename, but other filename is produced)
 - it is already call-by-value, so the outer filename is not changed anyway (no real need for const)

I'm not sure about this one. The thing is, we DO actually modify the filename. So I say it would be OK to have it not const. Of course, we could copy it internally, but is that really useful? The filename is already copied through the call-by-value semantic, so we would copy twice.